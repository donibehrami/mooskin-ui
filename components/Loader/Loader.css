.loader{
    display: flex;
    flex-direction: column;
    align-items: center;
    color: #8A0707;
}

.image{
    width: 170px;
    animation: two 2s infinite;
    animation-direction: alternate;
}

@keyframes two {
    0% {
        transform: rotateY(0);
    }
    100% {
        transform: rotateY(90deg);
    }
}

.hidden{
    display: none;
}