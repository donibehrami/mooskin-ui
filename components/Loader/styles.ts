import styled, {keyframes} from 'styled-components';

// Helpers
import { getNumberOrStringValue } from '../Box/styles';

// Models
import { ILoaderComponentProps } from './model';

// Components
import Box from '../Box/Box';

// "CSS" Variables
// import variables from '../_utils/globals/variables';

const spin = keyframes`
    0% { transform: rotate(0deg); }
    100% { transform: rotate(360deg); }
`;

export const StyledLoader = styled(Box)<ILoaderComponentProps>`
    border: ${(props) => getNumberOrStringValue(props.spinnerWidth)} solid #f3f3f3;
    border-top: ${(props) => getNumberOrStringValue(props.spinnerWidth)} solid #5ccdde;
    border-radius: 50%;
    width: ${(props) => getNumberOrStringValue(props.size)};
    height: ${(props) => getNumberOrStringValue(props.size)};
    animation: ${spin} 2s linear infinite;
`;

StyledLoader.displayName = 'StyledLoader';
