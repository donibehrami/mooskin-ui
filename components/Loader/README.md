# MooSkin UI - Loader Component

The MooSkin Loader Component is a out of the box loader to be used within the Col(Grid) component for automatic hide/show content depending on the status of the Loader.

___

### Usage

To start using the Loader Component first you have to Import it

```
Import {Loader} from '@moosend/mooskin';
```
or modular import
```
// JS
import Loader from '@moosend/mooskin/lib/Loader';

// CSS
import '@moosend/mooskin/lib/Loader/style.css';
```

And then you can simply start using it by typing

```
<Loader attribute1="atr" attribute2="asd" />
```

The `<Loader/>` component accepts different props, mostly used for customization.

### Examples


Pass it a custom loader

```
<Loader loader="some loader url" />
```

or pass it a custom animation

```
<Loader animation="animation class or style" />
```
<div class="playground-doc">

## Supported attributes

* `id` - id of the element
* `active` - wether the loader is active or not (this is a mandatory prop as it is required in relation with the Col(Grid) component)
* `loader` - custom loader image
* `animation` - custom animation
* `className` - css class
* `style` - loader style

</div>

Allthough these attributes are supported, all of them are optional.


#### For more

___

[MooSkin-UI](https://github.com/moosend/mooskin-ui)