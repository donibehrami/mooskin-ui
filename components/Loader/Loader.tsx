import * as React from 'react';

import styles from './Loader.css';

export interface ILoaderProps {

    /** id of the loader */
    id?: string;

    /** wether the loader is active or not */
    active: boolean;

    /** custom loader */
    loader?: string;

    /** custom animation */
    animation?: string | React.CSSProperties;

    /** override loader styles */
    style?: React.CSSProperties;

    /** override loader class */
    className?: string;

}

export default class Loader extends React.Component<ILoaderProps, {}> {

    static defaultProps = {
        className: '',
        style: {}
    };

    render() {

        const display = this.props.active ? '' : styles.hidden;

        const loader = this.props.loader ? this.props.loader :
                    'http://www.freeiconspng.com/uploads/load-icon-png-10.png';

        const animation = this.props.animation ? this.props.animation :
                        styles.animation;

        return (
            <div
                style={this.props.style}
                id={this.props.id}
                className={`loader-component ${styles.loader} ${display} ${this.props.className}`}
            >
                <img
                    src={loader}
                    className={`${styles.image} ${animation}`}
                />
            </div>
        );
    }
}
