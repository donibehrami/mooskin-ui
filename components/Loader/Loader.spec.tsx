import * as React from 'react';
import Loader from './Loader';

import { shallow } from 'enzyme';

describe('Loader', () => {
    test('renders correctly', () => {
        const tree = shallow(
            <Loader />
        );
        expect(tree).toMatchSnapshot();
    });
});
