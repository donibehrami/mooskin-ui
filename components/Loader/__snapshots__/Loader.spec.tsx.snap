// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders correctly 1`] = `
<Loader
  className=""
  size={50}
  spinnerWidth={5}
  style={Object {}}
>
  <Component
    className=""
    palette={
      Object {
        "backgroundColors": Object {
          "button": "#3fbaca",
          "buttonDisabled": "#e2e2e2",
          "common": "#ffffff",
          "main": "#f7f7f7",
          "secondaryToggle": "#ffffff",
          "switchDisabled": "#e7e7e7",
          "switchHandle": "#ffffff",
          "switchOff": "#9d9d9d",
          "switchOn": "#3fbaca",
          "tag": "#293346",
          "toggle": "#425579",
        },
        "borderColors": Object {
          "common": "#e2e2e2",
        },
        "fontColors": Object {
          "anchor": "#3fbaca",
          "button": "#ffffff",
          "common": "#293346",
          "description": "#9d9d9d",
          "disabledAnchor": "#9d9d9d",
          "label": "#2d2d2d",
          "secondaryToggle": "#9d9d9d",
          "tableHeader": "#425579",
          "tag": "#ffffff",
          "text": "#2d2d2d",
          "toggle": "#ffffff",
        },
      }
    }
    size={50}
    spinnerWidth={5}
    style={Object {}}
  >
    <StyledLoader
      className=""
      palette={
        Object {
          "backgroundColors": Object {
            "button": "#3fbaca",
            "buttonDisabled": "#e2e2e2",
            "common": "#ffffff",
            "main": "#f7f7f7",
            "secondaryToggle": "#ffffff",
            "switchDisabled": "#e7e7e7",
            "switchHandle": "#ffffff",
            "switchOff": "#9d9d9d",
            "switchOn": "#3fbaca",
            "tag": "#293346",
            "toggle": "#425579",
          },
          "borderColors": Object {
            "common": "#e2e2e2",
          },
          "fontColors": Object {
            "anchor": "#3fbaca",
            "button": "#ffffff",
            "common": "#293346",
            "description": "#9d9d9d",
            "disabledAnchor": "#9d9d9d",
            "label": "#2d2d2d",
            "secondaryToggle": "#9d9d9d",
            "tableHeader": "#425579",
            "tag": "#ffffff",
            "text": "#2d2d2d",
            "toggle": "#ffffff",
          },
        }
      }
      size={50}
      spinnerWidth={5}
      style={Object {}}
    >
      <Box
        className="sc-gtssRu bZjjdo"
        palette={
          Object {
            "backgroundColors": Object {
              "button": "#3fbaca",
              "buttonDisabled": "#e2e2e2",
              "common": "#ffffff",
              "main": "#f7f7f7",
              "secondaryToggle": "#ffffff",
              "switchDisabled": "#e7e7e7",
              "switchHandle": "#ffffff",
              "switchOff": "#9d9d9d",
              "switchOn": "#3fbaca",
              "tag": "#293346",
              "toggle": "#425579",
            },
            "borderColors": Object {
              "common": "#e2e2e2",
            },
            "fontColors": Object {
              "anchor": "#3fbaca",
              "button": "#ffffff",
              "common": "#293346",
              "description": "#9d9d9d",
              "disabledAnchor": "#9d9d9d",
              "label": "#2d2d2d",
              "secondaryToggle": "#9d9d9d",
              "tableHeader": "#425579",
              "tag": "#ffffff",
              "text": "#2d2d2d",
              "toggle": "#ffffff",
            },
          }
        }
        size={50}
        spinnerWidth={5}
        style={Object {}}
      >
        <Component
          className="sc-gtssRu bZjjdo"
          palette={
            Object {
              "backgroundColors": Object {
                "button": "#3fbaca",
                "buttonDisabled": "#e2e2e2",
                "common": "#ffffff",
                "main": "#f7f7f7",
                "secondaryToggle": "#ffffff",
                "switchDisabled": "#e7e7e7",
                "switchHandle": "#ffffff",
                "switchOff": "#9d9d9d",
                "switchOn": "#3fbaca",
                "tag": "#293346",
                "toggle": "#425579",
              },
              "borderColors": Object {
                "common": "#e2e2e2",
              },
              "fontColors": Object {
                "anchor": "#3fbaca",
                "button": "#ffffff",
                "common": "#293346",
                "description": "#9d9d9d",
                "disabledAnchor": "#9d9d9d",
                "label": "#2d2d2d",
                "secondaryToggle": "#9d9d9d",
                "tableHeader": "#425579",
                "tag": "#ffffff",
                "text": "#2d2d2d",
                "toggle": "#ffffff",
              },
            }
          }
          size={50}
          spinnerWidth={5}
          style={Object {}}
        >
          <StyledBox
            className="sc-gtssRu bZjjdo"
            palette={
              Object {
                "backgroundColors": Object {
                  "button": "#3fbaca",
                  "buttonDisabled": "#e2e2e2",
                  "common": "#ffffff",
                  "main": "#f7f7f7",
                  "secondaryToggle": "#ffffff",
                  "switchDisabled": "#e7e7e7",
                  "switchHandle": "#ffffff",
                  "switchOff": "#9d9d9d",
                  "switchOn": "#3fbaca",
                  "tag": "#293346",
                  "toggle": "#425579",
                },
                "borderColors": Object {
                  "common": "#e2e2e2",
                },
                "fontColors": Object {
                  "anchor": "#3fbaca",
                  "button": "#ffffff",
                  "common": "#293346",
                  "description": "#9d9d9d",
                  "disabledAnchor": "#9d9d9d",
                  "label": "#2d2d2d",
                  "secondaryToggle": "#9d9d9d",
                  "tableHeader": "#425579",
                  "tag": "#ffffff",
                  "text": "#2d2d2d",
                  "toggle": "#ffffff",
                },
              }
            }
            size={50}
            spinnerWidth={5}
            style={Object {}}
          >
            <div
              className="sc-bdnylx deSLUj sc-gtssRu bZjjdo"
              size={50}
              style={Object {}}
            />
          </StyledBox>
        </Component>
      </Box>
    </StyledLoader>
  </Component>
</Loader>
`;
